schema/representations.md

```diff
@@ -0,0 +1,256 @@
+
+Kinds and their Representations
+-------------------------------
+
+A type at the Schema level must be mapped onto a representation expressible
+within the Data Model.
+
+For all Schema kinds which have equivalents at the Data Model level, there is
+a default representation.  The default representation this is, broadly speaking,
+a no-op: a string remains a string, and a map is easy to enough to represent as,
+well, a map!
+
+For the perceived kinds (those introduced at the Schema level, with no direct
+equivalent in the Data Model), most also have default mappings: for example,
+structs will simply be mapped into maps; enum values are mapped into strings.
+Unions are a special case: there's no single widely agreed upon transcription
+pattern for unions from before the establishment of IPLD, so these require
+an explicit choice.
+
+Customized representation strategies are also possible.
+Each Schema type definition may specify such a custom representation, overriding
+the default.
+
+Representation strategies are identified by one primary keyword (which indicates
+what Kind in the Data Model will be transcribed to), and may contain several
+more detailed parameters which are specific to the strategy (e.g., the struct
+map representation has a concept of 'default' values which is unique to that
+strategy).
+
+### table
+
+(Kinds for which no meaningful customization is possible -- Null, Boolean,
+Integer, Float, String, Bytes, and List -- are elided.
+N.B., it's not ruled out that custom representations may be introduced for
+these kinds in the future; there are simply none to date.
+Link is also not described in this section, as it's a rather unique business.)
+
+- Map
+	- `map` representation -- the default -- transcribes to `map` in the Data Model.
+	- `stringpairs` representation -- transcribes to `string` in the Data Model.
+	- `listpairs` representation -- transcribes to `list` (of lists) in the Data Model.
+- Union
+	- `keyed` representation -- transcribes to a single-entry `map` in the Data Model.
+	- `kinded` representation -- transcribes to varying (!) kinds in the Data Model.
+	- `envelope` representation -- transcribes to a dual-entry `map` in the Data Model.
+	- `inline` representation -- transcribes to a `map` in the Data Model (and has additional limitations).
+- Struct
+	- `map` representation -- the default -- transcribes to `map` in the Data Model.
+	- `tuple` representation -- transcribes to `list` in the Data Model.
+	- `stringpairs` representation -- transcribes to `string` in the Data Model.
+	- `stringjoin` representation -- transcribes to `string` in the Data Model.
```

Comment:
same as above comment about `stringpairs`, it's pretty icky, do we have reason to believe it'll be useful as more than a novelty?